padle/padle.py

```diff
@@ -0,0 +1,801 @@
+import discord
+import datetime
+import random
+import asyncio
+import re
+import logging
+from io import BytesIO
+import json
+
+from typing import TYPE_CHECKING, Any, Union
+from redbot.core import Config, commands
+from math import ceil
+
+if TYPE_CHECKING:
+    from dbcog.dbcog import DBCog
+    # from dbcog.monster_graph import MonsterGraph # used for creating the filter
+
+from tsutils.tsubaki.links import MonsterImage, MonsterLink
+from tsutils.tsubaki.monster_header import MonsterHeader
+
+from tsutils.helper_functions import conditional_iterator
+from tsutils.user_interaction import get_user_confirmation, send_confirmation_message, send_cancellation_message, \
+    get_user_reaction
+from tsutils.emoji import NO_EMOJI, YES_EMOJI
+from tsutils.cogs.globaladmin import auth_check
+from redbot.core.utils.chat_formatting import box, pagify
+
+from tsutils.menu.view.closable_embed import ClosableEmbedViewState
+from tsutils.query_settings.query_settings import QuerySettings
+from padle.menu.closable_embed import ClosableEmbedMenu
+from padle.view.confirmation import PADleMonsterConfirmationView, PADleMonsterConfirmationViewProps
+from padle.menu.menu_map import padle_menu_map
+from padle.menu.padle_scroll import PADleScrollMenu, PADleScrollViewState
+from tsutils.menu.components.config import BotConfig
+from padle.monsterdiff import MonsterDiff
+
+from discordmenu.embed.components import EmbedThumbnail, EmbedMain, EmbedFooter
+from discordmenu.embed.view import EmbedView
+from tsutils.tsubaki.links import CLOUDFRONT_URL
+
+TSUBAKI_FLOWER_ICON_URL = CLOUDFRONT_URL + '/tsubaki/tsubakiflower.png'
+
+logger = logging.getLogger('red.padbot-cogs.padle')
+
+
+class PADle(commands.Cog):
+    """PADle"""
+
+    menu_map = padle_menu_map
+
+    def __init__(self, bot, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.bot = bot
+        self.config = Config.get_conf(self, identifier=94073)
+        self.config.register_user(todays_guesses=[], start=False, done=False, score=[],
+                                  edit_id="", channel_id="", all_guesses={})
```

Comment:
My guess is that edit_id and channel_id should be integers, not strings.  Maybe 0 would be a better default?